Doc/library/shutil.rst

```diff
@@ -259,6 +297,17 @@ Directory and files operations
    process.  :func:`ignore_patterns` can be used to create such a callable that
    ignores names based on glob-style patterns.
 
+   For example, this snippet uses the *ignore* argument to add a logging call::
+
+      from shutil import copytree
+      import logging
+
+      def _logpath(path, names):
+          logging.info('Working in %s', path)
+          return []   # nothing will be ignored
+
+      copytree(source, destination, ignore=_logpath)
+
```

Comment:
Taken in isolation, this example seems weird, as it's not actually ignoring anything. I'd suggest keeping the `ignore_patterns` example from the original text as well, and describe this example as something like:

> You don't have to ignore anything, if you simply want to run some code for every directory that gets processed. For example, this snippet uses the *ignore* argument to add a logging call::